control-plane/connect-inject/endpoints_controller_ent_test.go

```diff
@@ -2077,9 +2076,11 @@ func TestReconcileDeleteGatewayWithNamespaces(t *testing.T) {
 				require.False(t, resp.Requeue)
 
 				// After reconciliation, Consul should not have any instances of service-deleted.
-				serviceInstances, _, err := consulClient.Catalog().Service(consulSvcName, "", &api.QueryOptions{Namespace: ts.ConsulNS})
+				defaultNS, _, err := consulClient.Catalog().Service(consulSvcName, "", &api.QueryOptions{Namespace: "default"})
 				require.NoError(t, err)
-				require.Empty(t, serviceInstances)
```

Comment:
This fixes a "loophole" where the Mesh gateway could sneak through this check if it was not deregistered and the `ConsulNS` was not `"default"`.